### ISSUE TITLE
Online Status change doesn't trigger actions

### ISSUE THREAD

**Issue by aislanmaia** [redux-offline/redux-offline#146]
Just trying to setup my project with this great library... and great success so far, all is going well, but when deactivating my wifi connection (my attempt to simulating a offline state), make some inserts / updates, and activating it again, the actions doesn't get dispatched to the server unless I change a route or click something else that dispatch any another action, so then the actions (my update/insert while offline) get dispatched as expected. 

Is it intended to working like that, or there is something to do that get my actions dispatched as soon as my network status changes to `online`. 

Maybe the #132 issue is related.

**Comment by sorodrigo** [comment]
Could you share your offline config?

**Comment by birgernass** [comment]
I can reproduce the issue using react-native and version 2.3.3.
Nothing happens after I switch the internet connection back on. But as soon as I fire another redux action, the Offline/BUSY action gets dispatched as well. This is true for wifi and for mobile connections and especially troublesome if one uses the app in the subway etc.

```js
const offlineConfig = {
  ...defaultOfflineConfig,
  persistOptions: {
    blacklist: [/*...*/],
    storage: AsyncStorage,
  },
  effect,
}
```

I use a custom effectReconciler, which is just a thin wrapper around the fetch library.

**Comment by sorodrigo** [comment]
Thanks, we found that some store configs could cause this issue (#229). Are you using `offline(offlineConfig)` or `createOffline(offlineConfig)`?

**Comment by birgernass** [comment]
I'm using the latter, createOflline(offlineConfig).

**Comment by birgernass** [comment]
I switched enhanceStore and applyMiddleware and so far everything seems to work just fine. Is there something else I have to account for / that I need to change, because of the switch?

**Comment by sorodrigo** [comment]
Nope, it's just changing those two. I'm glad it work, this issue was quite annoying. Finally we can close it 😄 .

**Comment by aislanmaia** [comment]
Thanks for the replay @sorodrigo . @birgernass what you have to change to get it working again (as expected) ?

My configs is: 

```
/* Import redux-offline and default configs */
import { createOffline } from '@redux-offline/redux-offline';
import defaultConfig     from '@redux-offline/redux-offline/lib/defaults';


const config = {
  ...defaultConfig,
  effect: (effect, action) => {
	console.log(`Executing effect for ${action.type}`);
	return Api.send(effect, action);		
  },
};

const loggerMiddleware = createLogger({
  level: 'info',
  collapsed: true
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

// const { middleware, enhanceReducer, enhanceStore } = createOffline(defaultConfig);
const { middleware, enhanceReducer, enhanceStore } = createOffline(config);
const enhancedReducers = enhanceReducer(reducers);

export default function configureStore(browserHistory) {
  //const reduxRouterMiddleware = routerMiddleware(browserHistory);
  const createStoreWithMiddleware = composeEnhancers(
	applyMiddleware(
		//reduxRouterMiddleware,
		middleware,
		thunkMiddleware,
		loggerMiddleware,
	),
	enhanceStore,
)(createStore);

  // return createStoreWithMiddleware(reducers);
  return createStoreWithMiddleware(enhancedReducers);
}

```

**Comment by aislanmaia** [comment]
Ah, just to remember, I'm using the redux-offline version "^2.2.0" yet. So, I don't know if I need to update my package.json to current version. Any words ?

**Comment by sorodrigo** [comment]
Not necessarily, but whats blocking you from updating it?

You need to do `compose(enhanceStore, applyMiddleware)`.

**Comment by birgernass** [comment]
It really was that simple.

```diff
/* Import redux-offline and default configs */
import { createOffline } from '@redux-offline/redux-offline';
import defaultConfig     from '@redux-offline/redux-offline/lib/defaults';


const config = {
  ...defaultConfig,
  effect: (effect, action) => {
	console.log(`Executing effect for ${action.type}`);
	return Api.send(effect, action);		
  },
};

const loggerMiddleware = createLogger({
  level: 'info',
  collapsed: true
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

// const { middleware, enhanceReducer, enhanceStore } = createOffline(defaultConfig);
const { middleware, enhanceReducer, enhanceStore } = createOffline(config);
const enhancedReducers = enhanceReducer(reducers);

export default function configureStore(browserHistory) {
  //const reduxRouterMiddleware = routerMiddleware(browserHistory);
  const createStoreWithMiddleware = composeEnhancers(
+       enhanceStore,
	applyMiddleware(
		//reduxRouterMiddleware,
		middleware,
		thunkMiddleware,
		loggerMiddleware,
	)
-       enhanceStore,
)(createStore);

  // return createStoreWithMiddleware(reducers);
  return createStoreWithMiddleware(enhancedReducers);
}
```

**Comment by aislanmaia** [comment]
Haha oh my god, so simple change!!!! Tested and approved, it works!! Thank you @sorodrigo and @birgernass . My day is saved! :heart: 

@sorodrigo nothing is blocking me to update to current version, I just have not been following the recent activities in this repo. I will test an upgrade to 2.3.3 to see what happens. Is there really something important to get in terms of new features in the current version, in your opinion ? For what I remember now, in the 2.2 I've struggled to handle some batch or sequencial extra dispatches after commit and rollbacks using redux-thunk combined.
